### PR TITLE
Adds the Virus Extrapolator, removes Viral Base bottle

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
Virologists no longer rely on the Viral Base culture. Instead, they spawn with a robust tool called the Virus Extrapolator. The virus extrapolator is a normal-sized item, typically worn in the belt slot. it can be used to scan viruses contained in about anything that has them, and extract cultures. It can also extract single symptoms, though the level is limited by the level of the Extrapolator's scanning module. 

The extrapolator has a long cooldown on extractions, and takes a long time to do them. It can be upgraded by replacing the scanning module

other changes:
syringes retain diseases when used, and transfer them. Don't share syringes. Maint syringes have a chance to spawn with a disease already

several things will contain inert diseases that can be scanned by an extrapolator. The list is small, but may increase with time.

viro now has maint access, and the medical lathe can print scanning modules

## Why It's Good For The Game
instead of hiding in their autism fort all shift and begging for uranium, Virologists are encouraged to go around and hunt through maintenance  for their precious level 9 symptoms. The extraction feature of the extrapolator helps speed up virology a bit to compensate, and reduces the rampant rng factor, without being easily abuseable
## Changelog
:cl:
add: The disease extrapolator, a scanner that can scan or extract diseases and their symptoms, whatever they're in. it spawns in viro and the cmo locker, and virologists start with one
add: mice, vector, old bloodstains, crusty vomit, and used syringes can now contain extractable diseases. A few more minor sources also contain diseases to extract
add: sharing syringes spreads diseases
del: viro no longer spawns with highly potent viral base
tweak: viro now has maint access
tweak: medical lathe can print scanning modules
fix: fixed double-viro on box and meta. no idea when it got reverted, and why it was only on two maps...
imageadd: extrapolator sprites
tweak: maint is a bit grosser
/:cl:

### CODE REVIEW

**Comment by Dingo-Dongler** [comment]
So it can extrapolate directly from syringes? I feel like that should change to finally have a use for the viro monkeys.

**Comment by zeskorion** [comment]
> So it can extrapolate directly from syringes? I feel like that should change to finally have a use for the viro monkeys.

nah, monkeys are just for testing

File: code/datums/diseases/advance/presets.dm [hunk 1]
```diff
@@ -21,9 +21,9 @@
 	name = "Experimental Disease"
 	copy_type = /datum/disease/advance
 	var/randomname = TRUE
-	var/datum/symptom/specialsymptom = null
+	var/datum/symptom/setsymptom = null
 
-/datum/disease/advance/random/New(max_symptoms, max_level = 9)
+/datum/disease/advance/random/New(max_symptoms, max_level = 9, var/datum/symptom/specialsymptom = setsymptom)
```
Comment (PowerfulBacon) [inline code/datums/diseases/advance/presets.dm:26]:
```suggestion
/datum/disease/advance/random/New(max_symptoms, max_level = 9, datum/symptom/specialsymptom = setsymptom)
```

File: code/game/objects/effects/decals/cleanable/humans.dm [hunk 2]
```diff
@@ -19,11 +19,24 @@
...
 
 /obj/effect/decal/cleanable/blood/old/Initialize(mapload, list/datum/disease/diseases)
 	add_blood_DNA(list("Non-human DNA" = random_blood_type())) // Needs to happen before ..()
 	. = ..()
 	icon_state = "[icon_state]-old" //change from the normal blood icon selected from random_icon_states in the parent's Initialize to the old dried up blood.
+	if(prob(40))
+		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(1, 4), 2+(rand(1, 7)))
+		disease += R
+
+/obj/effect/decal/cleanable/blood/old/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/game/objects/effects/decals/cleanable/humans.dm:32]:
```suggestion
/obj/effect/decal/cleanable/blood/old/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

File: code/game/objects/effects/decals/cleanable/humans.dm [hunk 3]
```diff
@@ -118,12 +131,25 @@
...
 /obj/effect/decal/cleanable/blood/gibs/old/Initialize(mapload, list/datum/disease/diseases)
 	. = ..()
 	setDir(pick(1,2,4,8))
 	icon_state += "-old"
 	add_blood_DNA(list("Non-human DNA" = random_blood_type()))
+	if(prob(50))
+		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(1, 6), 4+(rand(1, 5)))
+		disease += R
+
+/obj/effect/decal/cleanable/blood/gibs/old/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/game/objects/effects/decals/cleanable/humans.dm:145]:
```suggestion
/obj/effect/decal/cleanable/blood/gibs/old/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)

```

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 4]
```diff
@@ -144,10 +144,23 @@
...
+	var/list/disease = list()
 
 /obj/effect/decal/cleanable/vomit/old/Initialize(mapload, list/datum/disease/diseases)
 	. = ..()
 	icon_state += "-old"
+	if(prob(90))//vomit is much more likely to be diseased than blood is
+		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(2, 6), 6+(rand(1, 3)))
+		disease += R
+
+/obj/effect/decal/cleanable/vomit/old/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/game/objects/effects/decals/cleanable/misc.dm:156]:
```suggestion
/obj/effect/decal/cleanable/vomit/old/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

**Comment by boodaliboo** [comment]
i may start playing bee just to mess with this

File: code/game/objects/items/devices/scanners.dm [hunk 5]
```diff
@@ -823,3 +823,133 @@ GENE SCANNER
...
+		var/datum/disease/advance/symptomholder = new 
+		symptomholder.name = chosen.name
+		symptomholder.symptoms += chosen
+		to_chat(user, "<span class='warning'>you begin isolating [chosen].</span>")
+		if(do_mob(user, AM, (600 / scanner.rating)))
+			create_culture(symptomholder, user)
+	else if(do_mob(user, AM, (timer / scanner.rating)))
+		create_culture(A, user)
+
+/obj/item/extrapolator/proc/create_culture(var/datum/disease/advance/A, mob/user)
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm:946]:
```suggestion
/obj/item/extrapolator/proc/create_culture(datum/disease/advance/A, mob/user)
```

File: code/game/objects/items/devices/scanners.dm [hunk 6]
```diff
@@ -823,3 +823,133 @@ GENE SCANNER
...
+			if(A.properties["stealth"] >= (2 + scanner.rating)) //the extrapolator can detect diseases of higher stealth than a normal scanner
+				continue
+			to_chat(user, "<span class='info'><font color='green'><b>[A.name]</b>, stage [A.stage]/5</color></span>")
+			to_chat(user, "<span class='info'><b>[A] has the following symptoms:</b></span>")
+			for(var/datum/symptom/S in A.symptoms)
+				to_chat(user, "<span class='info'>[S.name]</span>")
+		else
+			to_chat(user, "<span class='info'><font color='green'><b>[D.name]</b>, stage [D.stage]/[D.max_stages].</color></span>")
+
+/obj/item/extrapolator/proc/extrapolate(atom/AM, var/list/diseases = list(), mob/user, isolate = FALSE, timer = 200)
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm:916]:
```suggestion
/obj/item/extrapolator/proc/extrapolate(atom/AM, list/diseases = list(), mob/user, isolate = FALSE, timer = 200)
```

File: code/game/objects/items/devices/scanners.dm [hunk 7]
```diff
@@ -823,3 +823,133 @@ GENE SCANNER
...
+			if(locate(/datum/component/infective) in target.datum_components)
+				return //so the failure message does not show when we can actually extrapolate from a component
+			if(scan)
+				to_chat(user, "<span class='notice'>the extrapolator fails to return any data</span>")
+			else
+				to_chat(user, "<span class='notice'>the extrapolator's probe detects no diseases</span>")
+	else 
+		to_chat(user, "<span class='warning'>the extrapolator has no scanner installed</span>")
+
+/obj/item/extrapolator/proc/scan(atom/AM, var/list/diseases = list(), mob/user)
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm:902]:
```suggestion
/obj/item/extrapolator/proc/scan(atom/AM, list/diseases = list(), mob/user)
```

File: code/game/objects/items/devices/scanners.dm [hunk 8]
```diff
@@ -823,3 +823,133 @@ GENE SCANNER
...
+/obj/item/extrapolator/proc/extrapolate(atom/AM, var/list/diseases = list(), mob/user, isolate = FALSE, timer = 200)
+	var/list/advancediseases = list()
+	var/list/symptoms = list()
+	if(cooldown > world.time - (1200 / scanner.rating))
+		to_chat(user, "<span class='warning'>The extrapolator is still recharging!</span>")
+		return
+	for(var/datum/disease/advance/cantidate in diseases)
+		advancediseases += cantidate
+	if(!advancediseases.len)
+		to_chat(user, "<span class='warning'>There are no valid diseases to make a culture from.</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm:925]: Should have a return after this, so you aren't given an empty list and this message

File: code/game/objects/items/devices/scanners.dm [hunk 9]
```diff
@@ -823,3 +823,133 @@ GENE SCANNER
...
+	var/datum/disease/advance/A = input(user,"What disease do you wish to extract") in null|advancediseases
+	if(isolate)
+		for(var/datum/symptom/S in A.symptoms)
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm:929]: Is there any check if the user doesn't press anything, null.symptoms _might_ runtime here

File: code/game/objects/items/devices/scanners.dm [hunk 10]
```diff
@@ -823,3 +823,133 @@ GENE SCANNER
...
+
+/obj/item/extrapolator/proc/extrapolate(atom/AM, var/list/diseases = list(), mob/user, isolate = FALSE, timer = 200)
+	var/list/advancediseases = list()
+	var/list/symptoms = list()
+	if(cooldown > world.time - (1200 / scanner.rating))
+		to_chat(user, "<span class='warning'>The extrapolator is still recharging!</span>")
+		return
+	for(var/datum/disease/advance/cantidate in diseases)
+		advancediseases += cantidate
+	if(!advancediseases.len)
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm]:
```suggestion
	if(!LAZYLEN(advancediseases))
```

File: code/game/objects/items/devices/scanners.dm [hunk 11]
```diff
@@ -823,3 +823,133 @@ GENE SCANNER
...
+		to_chat(user, "<span class='warning'>There are no valid diseases to make a culture from.</span>")
+	var/datum/disease/advance/A = input(user,"What disease do you wish to extract") in null|advancediseases
+	if(isolate)
+		for(var/datum/symptom/S in A.symptoms)
+			if(S.level <= 6 + scanner.rating)
+				symptoms += S 
+			continue
+		var/datum/symptom/chosen = input(user,"What symptom do you wish to isolate") in null|symptoms
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm:933]: Make sure there is a check to see that chosen isn't null

**Comment by boodaliboo** [comment]
The ''syringes retain diseases'' sounds interesting but can lead to non stop infections. Maybe a way to sanitize them would help solve it

File: code/game/atoms.dm [hunk 12]
```diff
@@ -680,11 +680,19 @@
...
   */
 /atom/proc/teleport_act()
 	SEND_SIGNAL(src,COMSIG_ATOM_TELEPORT_ACT)
 
+/**
+  * Respond to our atom being checked by a virus extrapolator
+  *
+  * Default behaviour is to send COMSIG_ATOM_EXTRAPOLATOR_ACT and return FALSE
+  */
+/atom/proc/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/game/atoms.dm:693]:
```suggestion
/atom/proc/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

File: code/game/objects/structures/lavaland/necropolis_tendril.dm [hunk 13]
```diff
@@ -35,6 +36,17 @@ GLOBAL_LIST_INIT(tendrils, list())
 			M.ScrapeAway(null, CHANGETURF_IGNORE_AIR)
 	AddComponent(/datum/component/gps, "Eerie Signal")
 	GLOB.tendrils += src
+	var/datum/disease/advance/random/necropolis/R = new
+	necroseed += R
+
+/obj/structure/spawner/lavaland/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/game/objects/structures/lavaland/necropolis_tendril.dm:42]:
```suggestion
/obj/structure/spawner/lavaland/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

File: code/modules/mob/living/living_defense.dm [hunk 14]
```diff
@@ -403,3 +403,14 @@
 	..()
 	setMovetype(movement_type & ~FLOATING) // If we were without gravity, the bouncing animation got stopped, so we make sure we restart the bouncing after the next movement.
 
+/mob/living/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/living_defense.dm:406]:
```suggestion
/mob/living/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

File: code/modules/mob/living/simple_animal/friendly/hamster.dm [hunk 15]
```diff
@@ -43,6 +45,17 @@
 		vector_disease = new disease
 		message_admins("Vector was roundstart infected with [vector_disease.name]. Don't lynch the virologist!")
 		log_game("Vector was roundstart infected with [vector_disease.name].")
+	var/datum/disease/advance/R = new /datum/disease/advance/random(rand(1, 3))
+	extrapolatordisease += R
+
+/mob/living/simple_animal/pet/hamster/vector/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/simple_animal/friendly/hamster.dm:51]:
```suggestion
/mob/living/simple_animal/pet/hamster/vector/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 16]
```diff
@@ -35,6 +36,18 @@
 	icon_state = "mouse_[body_color]"
 	icon_living = "mouse_[body_color]"
 	icon_dead = "mouse_[body_color]_dead"
+	if(prob(20))
+		var/datum/disease/advance/R = new /datum/disease/advance/random(rand(2, 4))
+		ratdisease += R
+
+/mob/living/simple_animal/mouse/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm:43]:
```suggestion
/mob/living/simple_animal/mouse/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

File: code/modules/mob/living/simple_animal/hostile/macrophage.dm [hunk 17]
```diff
@@ -46,6 +46,16 @@
 	if(alreadyinfected)
 		return FALSE
 
+/mob/living/simple_animal/hostile/macrophage/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/simple_animal/hostile/macrophage.dm:49]:
```suggestion
/mob/living/simple_animal/hostile/macrophage/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

File: code/modules/mob/living/simple_animal/hostile/wumborian_fugu.dm [hunk 18]
```diff
@@ -138,3 +138,12 @@
 		A.environment_smash |= ENVIRONMENT_SMASH_STRUCTURES | ENVIRONMENT_SMASH_RWALLS
 		to_chat(user, "<span class='info'>You increase the size of [A], giving it a surge of strength!</span>")
 		qdel(src)
+
+/obj/item/fugu_gland/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/modules/mob/living/simple_animal/hostile/wumborian_fugu.dm:142]:
```suggestion
/obj/item/fugu_gland/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

File: code/modules/reagents/reagent_containers.dm [hunk 19]
```diff
@@ -154,3 +154,14 @@
 		filling.color = mix_color_from_reagents(reagents.reagent_list)
 		add_overlay(filling)
 	. = ..()
+
+/obj/item/reagent_containers/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = FALSE)
```
Comment (PowerfulBacon) [inline code/modules/reagents/reagent_containers.dm:158]:
```suggestion
/obj/item/reagent_containers/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = FALSE)
```

File: code/modules/reagents/reagent_containers/syringes.dm [hunk 20]
```diff
@@ -47,6 +48,15 @@
 /obj/item/reagent_containers/syringe/attackby(obj/item/I, mob/user, params)
 	return
 
+/obj/item/reagent_containers/syringe/extrapolator_act(mob/user, var/obj/item/extrapolator/E, scan = TRUE)
```
Comment (PowerfulBacon) [inline code/modules/reagents/reagent_containers/syringes.dm:51]:
```suggestion
/obj/item/reagent_containers/syringe/extrapolator_act(mob/user, obj/item/extrapolator/E, scan = TRUE)
```

File: code/game/objects/items/devices/scanners.dm [hunk 21]
```diff
@@ -823,3 +823,133 @@ GENE SCANNER
...
+	var/list/advancediseases = list()
+	var/list/symptoms = list()
+	if(cooldown > world.time - (1200 / scanner.rating))
+		to_chat(user, "<span class='warning'>The extrapolator is still recharging!</span>")
+		return
+	for(var/datum/disease/advance/cantidate in diseases)
+		advancediseases += cantidate
+	if(!advancediseases.len)
+		to_chat(user, "<span class='warning'>There are no valid diseases to make a culture from.</span>")
+	var/datum/disease/advance/A = input(user,"What disease do you wish to extract") in null|advancediseases
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm:927]: Not sure how this is meant to work, but when testing this input never actually showed up, even if there were 2 diseases on an object

**Comment by Ohagi-Chan** [comment]
Any chance the examine text of used syringes can potentially give away that they aren't fresh?

File: code/game/objects/items/devices/scanners.dm (hunk 21 above)
Comment (zeskorion) [reply code/game/objects/items/devices/scanners.dm:927]: Was it with a normal disease or an advanced disease? Extract doesnt work on non-advanced diseases

File: code/game/objects/items/devices/scanners.dm (hunk 10 above)
Comment (zeskorion) [reply code/game/objects/items/devices/scanners.dm]: doing this makes it fail to function

File: code/game/objects/items/devices/scanners.dm (hunk 9 above)
Comment (zeskorion) [reply code/game/objects/items/devices/scanners.dm:929]: added

File: code/game/objects/items/devices/scanners.dm (hunk 21 above)
Comment (zeskorion) [reply code/game/objects/items/devices/scanners.dm:927]: yep, works fine when tested

**Comment by zeskorion** [comment]
> Any chance the examine text of used syringes can potentially give away that they aren't fresh?

even the name does

File: code/game/objects/items/devices/scanners.dm [hunk 22]
```diff
@@ -823,3 +823,137 @@ GENE SCANNER
...
+	var/list/symptoms = list()
+	if(cooldown > world.time - (1200 / scanner.rating))
+		to_chat(user, "<span class='warning'>The extrapolator is still recharging!</span>")
+		return
+	for(var/datum/disease/advance/cantidate in diseases)
+		advancediseases += cantidate
+	if(!LAZYLEN(advancediseases))
+		to_chat(user, "<span class='warning'>There are no valid diseases to make a culture from.</span>")
+		return
+	var/datum/disease/advance/A = input(user,"What disease do you wish to extract") in null|advancediseases
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm:927]: There is still no check to see if A isn't null

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/scanners.dm (hunk 22 above)
Comment (zeskorion) [reply code/game/objects/items/devices/scanners.dm:927]: check is unecessary

**Comment by PowerfulBacon** [comment]
I'm on mobile and got nothing better to do, so after countless pings on discord I'll go through it one last time

**Comment by PowerfulBacon** [comment]
@ike709 can you rerun map diff bot?

File: code/game/objects/items/devices/scanners.dm [hunk 23]
```diff
@@ -823,3 +823,137 @@ GENE SCANNER
...
+		to_chat(user, "<span class='warning'>the extrapolator has no scanner installed</span>")
+
+/obj/item/extrapolator/proc/scan(atom/AM, var/list/diseases = list(), mob/user)
+	to_chat(user, "<span class='notice'><b>[src] detects the following diseases:</b></span>")
+	for(var/datum/disease/D in diseases)
+		if(istype(D, /datum/disease/advance))
+			var/datum/disease/advance/A = D
+			if(A.properties["stealth"] >= (2 + scanner.rating)) //the extrapolator can detect diseases of higher stealth than a normal scanner
+				continue
+			to_chat(user, "<span class='info'><font color='green'><b>[A.name]</b>, stage [A.stage]/5</color></span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm]: <font> tag is never closed

Comment (PowerfulBacon) [reply code/game/objects/items/devices/scanners.dm]: Font tag

File: code/game/objects/items/devices/scanners.dm [hunk 24]
```diff
@@ -823,3 +823,137 @@ GENE SCANNER
...
+			to_chat(user, "<span class='info'><b>[A] has the following symptoms:</b></span>")
+			for(var/datum/symptom/S in A.symptoms)
+				to_chat(user, "<span class='info'>[S.name]</span>")
+		else
+			to_chat(user, "<span class='info'><font color='green'><b>[D.name]</b>, stage [D.stage]/[D.max_stages].</color></span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm]: Font tag unenclosed

File: code/modules/antagonists/blob/blob_mobs.dm [hunk 25]
```diff
@@ -93,12 +93,25 @@
...
 	var/is_zombie = 0
 	gold_core_spawnable = HOSTILE_SPAWN
+	var/list/disease = list()
 
 /mob/living/simple_animal/hostile/blob/blobspore/Initialize(mapload, var/obj/structure/blob/factory/linked_node)
 	if(istype(linked_node))
 		factory = linked_node
 		factory.spores += src
 	. = ..()
+	/*var/datum/disease/advance/random/blob/R = new //either viro is cooperating with xenobio, or a blob has spawned and the round is probably over sooner than they can make a virus for this
```
Comment (PowerfulBacon) [inline code/modules/antagonists/blob/blob_mobs.dm:103]: Noooo!!!! You cant just have commented code no!!!! 😳

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Ok after these are changed I'd be relatively happy with this

File: code/modules/antagonists/blob/blob_mobs.dm (hunk 25 above)
Comment (zeskorion) [reply code/modules/antagonists/blob/blob_mobs.dm:103]: this shit didn't work, and i couldnt figure out how to fix it. Blob spores seemed to fuck getdiseaseid when spawned in this way, spawning cultures that would runtime and not work. Currently, blob spores have no disease by default, but contain the disease that spawned them if spawned by the blobspore symptom

**Review by PowerfulBacon — DISMISSED** [review]

File: code/game/objects/items/devices/scanners.dm [hunk 26]
```diff
@@ -906,12 +906,12 @@ GENE SCANNER
 			var/datum/disease/advance/A = D
 			if(A.properties["stealth"] >= (2 + scanner.rating)) //the extrapolator can detect diseases of higher stealth than a normal scanner
 				continue
-			to_chat(user, "<span class='info'><font color='green'><b>[A.name]</b>, stage [A.stage]/5</color></span>")
+			to_chat(user, "<span class='info'><font color='green'><b>[A.name]</b>, stage [A.stage]/5</b><</color></span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm]:
```suggestion
			to_chat(user, "<span class='info'><font color='green'><b>[A.name]</b>, stage [A.stage]/5</font></span>")

```

File: code/game/objects/items/devices/scanners.dm [hunk 27]
```diff
@@ -906,12 +906,12 @@ GENE SCANNER
...
 			if(A.properties["stealth"] >= (2 + scanner.rating)) //the extrapolator can detect diseases of higher stealth than a normal scanner
 				continue
-			to_chat(user, "<span class='info'><font color='green'><b>[A.name]</b>, stage [A.stage]/5</color></span>")
+			to_chat(user, "<span class='info'><font color='green'><b>[A.name]</b>, stage [A.stage]/5</b><</color></span>")
 			to_chat(user, "<span class='info'><b>[A] has the following symptoms:</b></span>")
 			for(var/datum/symptom/S in A.symptoms)
 				to_chat(user, "<span class='info'>[S.name]</span>")
 		else
-			to_chat(user, "<span class='info'><font color='green'><b>[D.name]</b>, stage [D.stage]/[D.max_stages].</color></span>")
+			to_chat(user, "<span class='info'><font color='green'><b>[D.name]</b>, stage [D.stage]/[D.max_stages].</b></color></span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/devices/scanners.dm]:
```suggestion
			to_chat(user, "<span class='info'><font color='green'><b>[D.name]</b>, stage [D.stage]/[D.max_stages].</font></span>")

```

**Review by PowerfulBacon — APPROVED** [review]
epic

**Pull request merged**